### PR TITLE
[videodb] cleanup files from removed media sources

### PR DESCRIPTION
**Pull request by mkortstiege**
In case the media source was removed without cleaning up the paths contents, there's no easy way to remove these files from the video library. Due to the fact the files still exist, the current cleanup logic will not remove them and you end up with old/unwanted entries.

This commit alters the video database cleanup logic to check whether or not the files belong to a valid and existing media source and prompts the user to decide how to proceed.

@Montellese for review please.

### CODE REVIEW

**Comment by mkortstiege** [comment]
jenkins build this please.

**Comment by mkortstiege** [comment]
Any objections?

File: xbmc/video/VideoDatabase.cpp [hunk 1]
```diff
@@ -8236,13 +8240,21 @@ std::vector<int> CVideoDatabase::CleanMediaType(const std::string &mediaType, co
...
+
+      if (sourceNotFound && sourcePath.empty())
+        sourcePath = parentPath;
+
       int sourcePathID = GetPathId(sourcePath);
       std::map<int, std::pair<bool, bool> >::const_iterator sourcePathsDeleteDecision = sourcePathsDeleteDecisions.find(sourcePathID);
       if (sourcePathsDeleteDecision == sourcePathsDeleteDecisions.end())
       {
-        bool sourcePathNotExists = !CDirectory::Exists(sourcePath, false);
+        bool sourcePathNotExists = (!CDirectory::Exists(sourcePath, false) || sourceNotFound);
```
Comment (Montellese) [inline xbmc/video/VideoDatabase.cpp]: The two conditions could be switched because `sourceNotFound` is much cheaper to evaluate.

**Comment by Montellese** [comment]
Looks good apart from my suggestion.

File: xbmc/video/VideoDatabase.cpp (hunk 1 above)
Comment (mkortstiege) [inline xbmc/video/VideoDatabase.cpp]: Thanks, will change.

**Comment by mkortstiege** [comment]
Updated and rebased.

**Comment by mkortstiege** [comment]
jenkins build and merge.

**Comment by mkortstiege** [comment]
Win32 build error is unrelated.

**Pull request merged**